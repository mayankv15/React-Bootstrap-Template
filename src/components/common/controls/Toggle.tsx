import React from "react";
import { Switch, Typography } from "@mui/material";
import { donationDetailsStyles } from "../styles/DonationDetailsStyles";
import { useTranslation } from "react-i18next";

interface Props {
  checked: boolean;
  onChange: (elm) => void;
  name?: string;
  label?: string;
  labelStyle?: string;
  disabled?: boolean;
}

const Toggle = ({
  checked,
  onChange,
  name,
  label,
  labelStyle,
  disabled,
}: Props) => {
  const classes = donationDetailsStyles();
  const { t } = useTranslation();

  return (
    <div style={{ display: "flex", flexDirection: "row" }}>
      <Switch
        classes={{
          root: classes.switchRoot,
          switchBase: classes.switchBase,
          thumb: classes.thumb,
          track: classes.track,
          checked: classes.checked,
        }}
        checked={checked}
        onChange={onChange}
        name={name}
        disabled={disabled}
      />
      {label && (
        <Typography className={labelStyle} style={{ paddingLeft: 2 }}>
          {t(label)}
        </Typography>
      )}
    </div>
  );
};

export default Toggle;
