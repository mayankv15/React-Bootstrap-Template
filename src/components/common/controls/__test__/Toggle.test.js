import React from "react";
import { render } from "@testing-library/react";
import Toggle from "../Toggle"

describe("Renders Toggle", () => {
    it("should render Toggle correctly", () => {
        const component = render(<Toggle
            checked={true}
            label="ON"
        />);
        expect(component).toMatchSnapshot();
    });
});
