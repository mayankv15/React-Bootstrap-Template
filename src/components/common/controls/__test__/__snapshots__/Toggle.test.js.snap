// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Renders Toggle should render Toggle correctly 1`] = `
Object {
  "asFragment": [Function],
  "baseElement": <body>
    <div>
      <div
        style="display: flex; flex-direction: row;"
      >
        <span
          class="MuiSwitch-root makeStyles-switchRoot-22"
        >
          <span
            aria-disabled="false"
            class="MuiButtonBase-root MuiIconButton-root PrivateSwitchBase-root-30 MuiSwitch-switchBase makeStyles-switchBase-23 MuiSwitch-colorSecondary PrivateSwitchBase-checked-31 Mui-checked makeStyles-checked-26"
          >
            <span
              class="MuiIconButton-label"
            >
              <input
                checked=""
                class="PrivateSwitchBase-input-33 MuiSwitch-input"
                type="checkbox"
                value=""
              />
              <span
                class="MuiSwitch-thumb makeStyles-thumb-24"
              />
            </span>
            <span
              class="MuiTouchRipple-root"
            />
          </span>
          <span
            class="MuiSwitch-track makeStyles-track-25"
          />
        </span>
        <p
          class="MuiTypography-root MuiTypography-body1"
          style="padding-left: 2px;"
        >
          ON
        </p>
      </div>
    </div>
  </body>,
  "container": <div>
    <div
      style="display: flex; flex-direction: row;"
    >
      <span
        class="MuiSwitch-root makeStyles-switchRoot-22"
      >
        <span
          aria-disabled="false"
          class="MuiButtonBase-root MuiIconButton-root PrivateSwitchBase-root-30 MuiSwitch-switchBase makeStyles-switchBase-23 MuiSwitch-colorSecondary PrivateSwitchBase-checked-31 Mui-checked makeStyles-checked-26"
        >
          <span
            class="MuiIconButton-label"
          >
            <input
              checked=""
              class="PrivateSwitchBase-input-33 MuiSwitch-input"
              type="checkbox"
              value=""
            />
            <span
              class="MuiSwitch-thumb makeStyles-thumb-24"
            />
          </span>
          <span
            class="MuiTouchRipple-root"
          />
        </span>
        <span
          class="MuiSwitch-track makeStyles-track-25"
        />
      </span>
      <p
        class="MuiTypography-root MuiTypography-body1"
        style="padding-left: 2px;"
      >
        ON
      </p>
    </div>
  </div>,
  "debug": [Function],
  "findAllByAltText": [Function],
  "findAllByDisplayValue": [Function],
  "findAllByLabelText": [Function],
  "findAllByPlaceholderText": [Function],
  "findAllByRole": [Function],
  "findAllByTestId": [Function],
  "findAllByText": [Function],
  "findAllByTitle": [Function],
  "findByAltText": [Function],
  "findByDisplayValue": [Function],
  "findByLabelText": [Function],
  "findByPlaceholderText": [Function],
  "findByRole": [Function],
  "findByTestId": [Function],
  "findByText": [Function],
  "findByTitle": [Function],
  "getAllByAltText": [Function],
  "getAllByDisplayValue": [Function],
  "getAllByLabelText": [Function],
  "getAllByPlaceholderText": [Function],
  "getAllByRole": [Function],
  "getAllByTestId": [Function],
  "getAllByText": [Function],
  "getAllByTitle": [Function],
  "getByAltText": [Function],
  "getByDisplayValue": [Function],
  "getByLabelText": [Function],
  "getByPlaceholderText": [Function],
  "getByRole": [Function],
  "getByTestId": [Function],
  "getByText": [Function],
  "getByTitle": [Function],
  "queryAllByAltText": [Function],
  "queryAllByDisplayValue": [Function],
  "queryAllByLabelText": [Function],
  "queryAllByPlaceholderText": [Function],
  "queryAllByRole": [Function],
  "queryAllByTestId": [Function],
  "queryAllByText": [Function],
  "queryAllByTitle": [Function],
  "queryByAltText": [Function],
  "queryByDisplayValue": [Function],
  "queryByLabelText": [Function],
  "queryByPlaceholderText": [Function],
  "queryByRole": [Function],
  "queryByTestId": [Function],
  "queryByText": [Function],
  "queryByTitle": [Function],
  "rerender": [Function],
  "unmount": [Function],
}
`;
